classes/Presenter/Cart/CartPresenter.php

```diff
@@ -21,138 +21,56 @@
 namespace PrestaShop\Module\PrestashopCheckout\Presenter\Cart;
 
 use PrestaShop\Module\PrestashopCheckout\Presenter\PresenterInterface;
-use PrestaShop\Module\PrestashopCheckout\ShopContext;
-use PrestaShop\PrestaShop\Adapter\Cart\CartPresenter as PsCartPresenter;
 
 /**
  * Present the cart waiting by the create order paypal builder
  */
 class CartPresenter implements PresenterInterface
 {
-    /**
-     * @var \Cart
-     */
-    private $cart;
-
-    /**
-     * @var \Customer
-     */
-    private $customer;
-
-    /**
-     * @var \Language
-     */
-    private $language;
-
-    public function __construct(\Context $context)
-    {
-        $this->setCart($context->cart);
-        $this->setCustomer($context->customer);
-        $this->setLanguage($context->language);
-    }
-
     /**
      * Present improved cart
      *
      * @return array
+     *
+     * @throws \Exception
      */
     public function present()
     {
-        $productList = $this->getCart()->getProducts();
+        $context = \Context::getContext();
+        $productList = $context->cart->getProducts();
 
-        $cart = (array) $this->getCart();
+        $cart = (array) $context->cart;
 
-        if ((new ShopContext())->isShop17()) {
-            $cart = new PsCartPresenter();
-            $cart = $cart->present($this->getCart());
+        if (class_exists('\PrestaShop\PrestaShop\Adapter\Cart\CartPresenter')) {
```

Comment:
I'm curious, why have you chosen this test ?

Reply:
Avoid to create another instance of ShopContext not required because we could only check if class is declared
In v2.0.0 we don't use CartPresenter anymore

Reply:
And the `use` at beginning of files of a class not exist on 1.6 can be a problem, that's why I removed it in favor of full namespace